synapse/handlers/sync.py

```diff
@@ -174,37 +194,52 @@ def full_state_sync(self, sync_config, timeline_since_token):
         """
         now_token = yield self.event_sources.get_current_token()
 
-        now_token, ephemeral_by_room = yield self.ephemeral_by_room(
-            sync_config, now_token
-        )
+        if sync_config.is_guest:
+            room_list = [
+                GuestRoom(room_id, Membership.JOIN)
+                for room_id in sync_config.filter.list_rooms()
+            ]
 
-        presence_stream = self.event_sources.sources["presence"]
-        # TODO (mjark): This looks wrong, shouldn't we be getting the presence
-        # UP to the present rather than after the present?
-        pagination_config = PaginationConfig(from_token=now_token)
-        presence, _ = yield presence_stream.get_pagination_rows(
-            user=sync_config.user,
-            pagination_config=pagination_config.get_source_config("presence"),
-            key=None
```

Comment:
This is sort of fixing the TODO. The presence event stream has extremely similar implementations for get_new_events and get_pagination_rows. Switching to using get_new_events gives me more confidence that it will actually return the events we want it to return, since it should now give the same events that the client would have received in an incremental sync.

The main reason for switching it now is because I can supply a list of room_ids so that it can be used for guest access.